main.js

```diff
@@ -1006,6 +1015,75 @@ class Unifi extends utils.Adapter {
         return data;
     }
 
+    /**
+     * Function to switch poe power for port of device
+     * @param {String} site
+     * @param {String} deviceMac
+     * @param {String} port
+     * @param {Boolean} val
+     */
+    async switchPoeOfPort(site, deviceMac, port, val) {
+        try {
+            this.log.info(`switchPoeOfPort: switching poe power of port ${port} for device ${deviceMac} to ${val}`);
+
+            // we have to get whole data of 'port_overrides' to change poe power of single port.
+            // we must sent the 'port_overrides' for all ports, otherwise the other port will set to default settings
+
+            const result = await this.fetchDevices(site);
+
+            const dataDevice = result.filter(x => x.mac === deviceMac);
+
+            if (dataDevice && dataDevice.length) {
+                const deviceId = dataDevice[0].device_id;
+
+                // eslint-disable-next-line prefer-const
+                let port_overrides = dataDevice[0].port_overrides;
+
+                if (port_overrides && port_overrides.length > 0) {
+                    const indexOfPort = port_overrides.findIndex(x => x.port_idx === parseInt(port));
+
+                    if (indexOfPort > 0) {
```

Comment:
0 is a valid index. needs to be
`if (indexOfPort "= -1) {`

Reply:
Thanks for reviewing it. Your are absolutly right, I will change this.